python-frontend/src/main/java/org/sonar/python/TokenLocation.java

```diff
@@ -36,9 +36,13 @@ public TokenLocation(Token token) {
     String[] lines = value.split("\r\n|\n|\r", -1);
 
     if (lines.length > 1) {
-      endLine = token.line() + lines.length - 1;
-      endLineOffset = lines[lines.length - 1].length();
-
+      if (token.isCompressed()) {
+        endLine = token.line();
+        endLineOffset = this.startLineOffset + token.valueLength();
+      } else {
+        endLine = token.line() + lines.length - 1;
+        endLineOffset = lines[lines.length - 1].length();
```

Comment:
Not sure when this happens but shouldn't we just compute as well the `valueLength()` of the last line when uncompressed?